### ISSUE TITLE
Use socket.io again to provide Safari support?

### ISSUE THREAD

**Issue by ericfranz** [OSC/ood-shell#39]
When we stopped using wetty we also stopped using socket.io. Ends up **socket.io falls back to polling when web sockets are not available**. 

Installing this commit 2883842093f65fe0812e419a0494ec827979b316 as a dev app and building results in a working shell from Safari on ood.osc.edu.

In Safari polling results in the terminal being slow, but if you make the window smaller it is usable, sort've.

<img width="1084" alt="170715_231504" src="https://user-images.githubusercontent.com/512333/28244255-7d391b0a-69b3-11e7-8d29-c9f1f2767055.png">

**Comment by ericfranz** [comment]
Ha actually the socket.io.js version just works. You don't need to force it to use polling, it auto-falls back.

**Comment by nickjer** [comment]
A few things I worry about:

- Back when we used Socket.io it was not possible to add the necessary `X-Requested-With: XMLHttpRequest` header (needed by mod_auth_openidc) to the AJAX request without doing some crazy monkey-patching (we patched the core library `XMLHttpRequest`).

  This is still do-able but if we carry over the client to the Dashboard, this may or may not break any tools (or future tools) that rely on `XMLHttpRequest` in the Dashboard.

- By introducing polling we may open ourselves up to providing support and maintenance for a new class of problems. For example:
  - What happens when the Apache proxy session times out (~ 8 hours I think), the terminal will just stop working (not an issue with websockets because a one-time connection is established and maintained even when the user is logged out)
  - What happens when Trey restarts the PUN's with no active connections, the shell app will shutdown if there isn't an active AJAX request at that time killing the user's session (not an issue with websockets because websockets maintain the connection indefinitely and aren't intended to be RESTful)
